werkzeug/urls.py

```diff
@@ -449,6 +449,29 @@ def url_parse(url, scheme=None, allow_fragments=True):
     return result_type(scheme, netloc, url, query, fragment)
 
 
+def _make_url_encoder(charset='utf-8', errors='strict', safe='/:', unsafe=''):
+    if isinstance(safe, text_type):
+        safe = safe.encode(charset, errors)
+    if isinstance(unsafe, text_type):
+        unsafe = unsafe.encode(charset, errors)
+    safe = (frozenset(safe) | frozenset(_always_safe)) - frozenset(unsafe)
+    if not isinstance(next(iter(safe)), int):
+        safe = frozenset(map(ord, safe))
+    table = [chr(c) if c in safe else '%%%02X' % c for c in range(256)]
+    quote = lambda s: ''.join(map(table.__getitem__, s))
+    if isinstance(''.encode(), str):
+        return lambda s: quote(bytearray(s))
+    return quote
+
+
+fast_url_quote = _make_url_encoder()
```

Comment:
Just checking, did you benchmark this to confirm it's actually faster? If it is faster and does the same thing as `url_quote`, why not replace `url_quote` completely?